pkg/file/path.go

```diff
@@ -83,3 +81,20 @@ func (p Path) AllPaths() []Path {
 	fullPaths[len(parents)] = p
 	return fullPaths
 }
+
+// ConstituentPaths returns all valid constituent paths (e.g. /home/wagoodman/file.txt -> /, /home, /home/wagoodman )
+func (p Path) ConstituentPaths() []Path {
+	parents := strings.Split(strings.Trim(string(p), DirSeparator), DirSeparator)
+	fullPaths := make([]Path, len(parents))
+	for idx := range parents {
+		cur := DirSeparator + strings.Join(parents[:idx], DirSeparator)
+		fullPaths[idx] = Path(cur)
+	}
+	return fullPaths
```

Comment:
It'd be good to deduplicate this logic, given the implementation for `AllPaths`. One suggestion would be to have `AllPaths` consume this method and append the original path.

Reply:
I meant to do exactly that, I'll adjust